agent-control/src/sub_agent/on_host/supervisor.rs

```diff
@@ -208,159 +193,171 @@ impl NotStartedSupervisorOnHost {
         )
     }
 
-    fn start_process_thread(
+    fn start_process_threads(
```

Comment:
this method is something like 200 lines 😕 
Could we "break it down"? I feel like it is doing way too much